openrelik_worker_common/mount_utils.py

```diff
@@ -0,0 +1,309 @@
+
+        process = subprocess.run(
+            blkid_command, capture_output=True, check=False, text=True
+        )
+        if process.returncode == 0:
+            return process.stdout.strip()
+        else:
+            logger.error(
+                f"Error running blkid on {devname}: {process.stderr} {process.stdout}"
+            )
+            raise RuntimeError(
+                f"Error running blkid on {devname}: {process.stderr} {process.stdout}"
+            )
+
+    def mount(self, partition_name: str = ""):
+        """Mounts a disk or one or more partititions on a mountpoint.
+
+        Args:
+            partitions_name (str): Name of specific partition to mount.
+
+        Returns:
+            list: A list of paths the disk/partitions have been mounted on.
+
+        Raises:
+          RuntimeError: If there as an error running mount.
+        """
+        to_mount = []
+
+        if partition_name and partition_name not in self.partitions:
+            logger.error(
+                f"Error running mount: partition name {partition_name} not found"
+            )
+            raise RuntimeError(
+                f"Error running mount: partition name {partition_name} not found"
+            )
+
+        if partition_name:
+            to_mount.append(partition_name)
+        elif not self.partitions:
+            to_mount.append(self.blkdevice)
+        elif self.partitions:
+            to_mount = self.partitions
+
+        if not to_mount:
+            logger.error(f"Error: nothing to mount")
+            raise RuntimeError(f"Error: nothing to mount")
+
+        for mounttarget in to_mount:
+            logger.info(f"Trying to mount {mounttarget}")
+            mount_command = ["sudo", "mount"]
+            fstype = self._get_fstype(mounttarget)
+            if fstype == "xfs":
+                mount_command.extend(["-o", "ro,norecovery"])
+            elif fstype in ["ext2", "ext3", "ext4"]:
+                mount_command.extend(["-o", "ro,noload"])
+            else:
+                mount_command.extend(["-o", "ro"])
+
+            mount_command.append(mounttarget)
+
+            mount_folder = f"{self.mountroot}/{uuid4().hex}"
+            os.makedirs(mount_folder)
+
+            mount_command.append(mount_folder)
+
+            process = subprocess.run(
+                mount_command, capture_output=True, check=False, text=True
+            )
+            if process.returncode == 0:
+                logger.info(f"Mounted {mounttarget} to {mount_folder}")
+                self.mountpoints.append(mount_folder)
+            else:
+                logger.error(
+                    f"Error running mount on {mounttarget}: {process.stderr} {process.stdout}"
+                )
+                raise RuntimeError(
+                    f"Error running mount on {mounttarget}: {process.stderr} {process.stdout}"
+                )
+        return self.mountpoints
+
+    def umount(self):
```

Comment:
Don't we always want to detatch (destroy()) the blkdevice when we have umounted? Now it is up to the caller to use destoy() to make both happen. 

Perhaps we can wrap both of these in a cleanup() function:
```
def umount(self):
    self._umount_all()
    self._detach_device()

def _umount_all(self). # Umounts all registered mount_points
def _detach_device(self). # Detaches the loop device

```

Just a thought.